core/os/safe_binary_mutex.h

```diff
@@ -93,16 +93,44 @@ template <int Tag>
 class MutexLock<SafeBinaryMutex<Tag>> {
 	friend class ConditionVariable;
 
-	THREADING_NAMESPACE::unique_lock<THREADING_NAMESPACE::mutex> lock;
+	mutable THREADING_NAMESPACE::unique_lock<THREADING_NAMESPACE::mutex> _lock;
 
 public:
 	_ALWAYS_INLINE_ explicit MutexLock(const SafeBinaryMutex<Tag> &p_mutex) :
-			lock(p_mutex.mutex) {
+			_lock(p_mutex.mutex) {
 		SafeBinaryMutex<Tag>::count++;
-	};
+	}
+
 	_ALWAYS_INLINE_ ~MutexLock() {
 		SafeBinaryMutex<Tag>::count--;
```

Comment:
This implies that `MutexLock` must be locked when it hits the destructor, is this intentional? If yes, I think there should be a note somewhere about it and `DEV_ASSERT(SafeBinaryMutex<Tag>::count)` or something like this in the destructor

Reply:
There is, in the Mutex destructor, the lock isn't responsible for this

Reply:
But regardless this PR didn't introduce this particular code so I think if there's some issue with the structure here it'd be best to open an issue and make a dedicated fix/improvement

Reply:
> There is, in the Mutex destructor, the lock isn't responsible for this, it's only when the Mutex is destroyed that this should be checked

I meant `DEV_ASSERT(SafeBinaryMutex<Tag>::count)` before `SafeBinaryMutex<Tag>::count--`, so there wouldn't be negative numbers

Reply:
That's a good point, will look

Reply:
Added that and an additional improvement as a separate commit